### DISCUSSION TITLE
gis

### DISCUSSION THREAD

**Discussion by jinyuttt** [AvaloniaUI/Avalonia#5644] [Q&A]
I want to develop secondary development based on osgeearth. I don't know how to integrate with AvaloniaUI . Is there an example? Is it feasible?

**Comment by maxkatz6** [comment]
You need to use OpenGL based SDK and integrate it with Avalonia.
For that you need to use OpenGlControlBase, we have an example in the ControlCatalog for it - https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/OpenGlPage.xaml.cs

Note it's not stable on Windows right now, and you will probably need to enable UseWgl in the Win32PlatformOptions.
